rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -570,7 +596,19 @@ public void onRemoval(RemovalNotification<String, FileMetadata> notification)
       FileMetadata md = notification.getValue();
       try {
         md.closeAndCleanup(notification.getCause(), fileMetadataCache);
-        cacheEvictionCount.inc();
+        switch (notification.getCause()) {
+          case EXPLICIT:
+            cacheInvalidationCount.inc();
+            break;
+          case SIZE:
+            cacheEvictionCount.inc();
```

Comment:
We should differentiate SIZE from EXPIRED, users would be more interested in eviction count due to SIZE

Reply:
But user can change the default. If you plan to get rid of count in case of EXPIRED case then this is fine but I would then suggest to rename the metric a bit to show it is for SIZE based expiry.

Reply:
Dont need to do it now if it needs more changes than just rename, in that case can take it up separately

Reply:
Done